src/checks/#cc4a#check_in_iteration.clas.abap

```diff
@@ -0,0 +1,347 @@
+class /cc4a/check_in_iteration definition
+public
+  final
+  create public.
+  public section.
+    interfaces if_ci_atc_check.
+    constants: begin of quickfix_codes,
+                 if_quickfix    type cl_ci_atc_quickfixes=>ty_quickfix_code value `C_I_I_IF`,
+                 where_quickfix type cl_ci_atc_quickfixes=>ty_quickfix_code value `C_I_I_LOOP`,
+               end of quickfix_codes.
+
+    constants:
+      begin of pseudo_comment,
+        check_in_iteration type string value 'CHECK_IN_ITERATION',
+      end of pseudo_comment.
+
+    constants:
+      begin of finding_codes,
+        check_in_iteration type if_ci_atc_check=>ty_finding_code value 'C_I_I',
+      end of finding_codes.
+
+    methods constructor.
+  protected section.
+  private section.
+    data code_provider     type ref to if_ci_atc_source_code_provider.
+    data assistant_factory type ref to cl_ci_atc_assistant_factory.
+    data meta_data         type ref to /cc4a/if_check_meta_data.
+
+    types: begin of negate_structure,
+             use_not               type abap_bool,
+             comperasion_operators type standard table of i with non-unique default key,
```

Comment:
Typo
```suggestion
             comparison_operators type standard table of i with non-unique default key,
```
Also: Don't use `with default key` - if you want to express this is just an array of integers, say: `with empty key`